share/eds/retriever_quadrant.go

```diff
@@ -93,31 +89,16 @@ func newQuadrants(dah *da.DataAvailabilityHeader) []*quadrant {
 	return quadrants
 }
 
-// index calculates index for a share in a data square slice flattened by rows.
-//
-// NOTE: The complexity of the formula below comes from:
-//   - Goal to avoid share copying
-//   - Goal to make formula generic for both rows and cols
-//   - While data square is flattened by rows only
-//
-// TODO(@Wondertan): This can be simplified by making rsmt2d working over 3D byte slice(not
-// flattened)
-func (q *quadrant) index(rootIdx, cellIdx int) int {
-	size := len(q.roots)
-	// half square offsets, e.g. share is from Q3,
-	// so we add to index Q1+Q2
-	halfSquareOffsetCol := pow(size*2, q.source)
-	halfSquareOffsetRow := pow(size*2, q.source^1)
-	// offsets for the axis, e.g. share is from Q4.
-	// so we add to index Q3
-	offsetX := q.x * halfSquareOffsetCol * size
-	offsetY := q.y * halfSquareOffsetRow * size
-
-	rootIdx *= halfSquareOffsetRow
-	cellIdx *= halfSquareOffsetCol
-	return rootIdx + cellIdx + offsetX + offsetY
-}
-
-func pow(x, y int) int {
-	return int(math.Pow(float64(x), float64(y)))
+// pos calculates position of a share in a data square.
+func (q *quadrant) pos(rootIdx, cellIdx int) (int, int) {
+	cellIdx += len(q.roots) * q.x
+	rootIdx += len(q.roots) * q.y
+	switch q.source {
+	case rsmt2d.Row:
+		return rootIdx, cellIdx
+	case rsmt2d.Col:
+		return cellIdx, rootIdx
+	default:
+		panic("unknown axis")
+	}
```

Comment:
nice, much cleaner

Reply:
This was honestly my main motivation to make this PR 🙂